src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -174,8 +177,20 @@ public ServerStore(RavenConfiguration configuration, RavenServer server)
                            Uri.TryCreate(Configuration.Core.ServerUrls[0], UriKind.Absolute, out var uri) == false ||
                            uri.Port != 0;
 
-            var maxConcurrentBackupsTasks = Configuration.Backup.MaxNumberOfConcurrentBackups ?? int.MaxValue;
-            ConcurrentBackupsSemaphore = new SemaphoreSlim(maxConcurrentBackupsTasks);
+            _skipChangingMaxConcurrentBackups = Configuration.Backup.MaxNumberOfConcurrentBackups != null;
+            GlobalBackupConfiguration = new GlobalBackupConfiguration(Configuration.Backup.MaxNumberOfConcurrentBackups ?? int.MaxValue);
+        }
+
+        public void UpdateMaxConcurrentBackups(int licensedCores)
+        {
+            if (_skipChangingMaxConcurrentBackups)
+                return;
+
+            var maxNumberOfConcurrentBackups = Math.Max(licensedCores / 2, 1);
```

Comment:
Don't we need to account for the total number of cores that the machine has?
User may have 16 cores license, but running on 2 core machine in a cluster of 8 machines.

Reply:
The name is misleading. It should be `utilizedCores`.